File: classStack.py
class Stack:
    def __init__(self,c):
        self.capacidade = c
        self.front,self.rear = 0,0
        self.stack = []
        
        return self.stack
    
    def push(self,x):
        if self.rear == self.capacidade:
            print("Erro StackOverflow")
        else:
            self.rear += 1
            self.stack.append(x)
        return self.stack
    
    
    def pop(self):
        if self.rear == 0:
            print("Erro Empty Stack")
        else:
            self.stack.pop()
            self.rear -= 1
        return self.stack
    
    def length(self):
        return self.rear
    
    def getHead(self):
        return self.stack[self.front]
    
    def __repr__(self):
        s = '['
        #print("[",end="")
        for i in self.stack:
            s += str(i)
            if not(i == self.stack[-1]):
                s += ','
        s += "]"
        return s

File: classStack.js
class Stack {
  
  constructor(c) {
    globalThis.capacidade = c
    globalThis.stack = new Array(self.capacidade) 
    //inicalizando um vetor vazio
    globalThis.inicio = 0  
    globalThis.fim = 0    
    return globalThis.stack;
  }
  
   insert (x) {
    if (globalThis.fim == globalThis.capacidade) {
      return "Stack Overflow";
    }
    else {
      globalThis.fim += 1
      globalThis.stack.push(x)
      return globalThis.stack;
    }
  }

   delete () {
    if (globalThis.fim == 0) {
      return "Empty Stack";
    }
    else {
      let a = globalThis.stack[globalThis.fim];
      globalThis.fim -= 1
      globalThis.stack.pop()
      return globalThis.stack , a;
    }
  }


   len(){
   return globalThis.fim;
  }
  
   /*print(){
    let i;
    for (i of globalThis.stack) {
      document.write(i);
    }
  }*/
  
}
